static/js/containers/PostPage.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import R from 'ramda';
+
+import Card from '../components/Card';
+import Loading, { anyProcessing, allLoaded } from '../components/Loading';
+import ChannelBreadcrumbs from '../components/ChannelBreadcrumbs';
+import PostDisplay from '../components/PostDisplay';
+
+import { actions } from '../actions';
+
+import type { Dispatch } from 'redux';
+import type { Match } from 'react-router';
+
+class PostPage extends React.Component {
+  props: {
+    match:            Match,
+    dispatch:         Dispatch,
+    posts:            Object,
+    channels:         Object,
+  };
+
+  getMatchParams = () => {
+    const { match: { params }} = this.props;
+    return [params.postID, params.channelName];
+  }
+
+  updateRequirements = () => {
+    const { dispatch, channels, posts } = this.props;
+    const [postID, channelName] = this.getMatchParams();
+
+    if (R.isNil(posts.data.get(postID))) {
```

Comment:
Do we need these checks here? It seems like we only get to this function if the request already completed successfully